python/python/glide/async_commands/sorted_set.py

```diff
@@ -169,6 +170,145 @@ def __init__(
+        latitude (float): The latitude coordinate.
+    """
+
+    def __init__(self, longitude: float, latitude: float):
+        self.longitude = longitude
+        self.latitude = latitude
+
+
+class GeoUnit(Enum):
+    """
+    Enumeration representing distance units options for the `GEODIST` command.
+    """
+
+    METERS = "m"
+    """
+    Represents distance in meters.
+    """
+    KILOMETERS = "km"
+    """
+    Represents distance in kilometers.
+    """
+    MILES = "mi"
+    """
+    Represents distance in miles.
+    """
+    FEET = "ft"
+    """
+    Represents distance in feet.
+    """
+
+
+class OrderBy(Enum):
+    """
+    Enumeration representing sorting order options for `GEOSEARCH` command.
+    """
+
+    ASCENDING = "ASC"
+    """
+    Sort returned items from the nearest to the farthest, relative to the center point.
+    """
+    DESCENDING = "DESC"
+    """
+    Sort returned items from the farthest to the nearest, relative to the center point.
+    """
+
+
+class GeoSearchByRadius:
+    """
+    Represents search criteria of searching within a certain radius from a specified point.
+
+    Args:
+        radius (float): Radius of the search area.
+        unit (GeoUnit): Unit of the radius. See `GeoUnit`.
+    """
+
+    def __init__(self, radius: float, unit: GeoUnit):
+        """
+        Initialize the search criteria.
+        """
+        self.radius = radius
+        self.unit = unit
+
+    def __str__(self) -> str:
+        """
+        Convert the search criteria to the corresponding part of the Redis command.
+
+        Returns:
+            str: String representation of the search criteria.
+        """
+        return f"BYRADIUS {self.radius} {self.unit.value}"
+
+
+class GeoSearchByBox:
+    """
+    Represents search criteria of searching within a specified rectangular area.
+
+    Args:
+        width (float): Width of the bounding box.
+        height (float): Height of the bounding box
+        unit (GeoUnit): Unit of the radius. See `GeoUnit`.
+    """
+
+    def __init__(self, width: float, height: float, unit: GeoUnit):
+        """
+        Initialize the search criteria.
+        """
+        self.width = width
+        self.height = height
+        self.unit = unit
+
+    def __str__(self) -> str:
+        """
+        Convert the search criteria to the corresponding part of the Redis command.
+
+        Returns:
+            str: String representation of the search criteria.
+        """
+        return f"BYBOX {self.width} {self.height} {self.unit.value}"
+
+
+class GeoSearchCount:
+    """
+    Represents the count option for limiting the number of results in a GeoSearch.
+
+    Args:
+        count (int): The maximum number of results to return.
+        any_option (bool): Whether to allow returning as enough matches are found.
+        This means that the results returned may not be the ones closest to the specified point. Default to False.
```

Comment:
Redis docs for `ANY` isn't trivial. As I understand, server may include to the response a value which doesn't fit the search criteria (out of the radius, for example), but easy to calculate.
Maybe reword this doc.

Reply:
```
redis> GEOSEARCH Sicily FROMLONLAT 15 37 BYBOX 400 400 km ASC WITHCOORD WITHDIST
1) 1) "Catania"
   2) "56.4413"
   3) 1) "15.08726745843887329"
      2) "37.50266842333162032"
2) 1) "Palermo"
   2) "190.4424"
   3) 1) "13.36138933897018433"
      2) "38.11555639549629859"
3) 1) "edge2"
   2) "279.7403"
   3) 1) "17.24151045083999634"
      2) "38.78813451624225195"
4) 1) "edge1"
   2) "279.7405"
   3) 1) "12.7584877610206604"
      2) "38.78813451624225195"
redis> GEOSEARCH Sicily FROMLONLAT 15 37 BYBOX 400 400 km ASC WITHCOORD WITHDIST count 1
1) 1) "Catania"
   2) "56.4413"
   3) 1) "15.08726745843887329"
      2) "37.50266842333162032"
redis> GEOSEARCH Sicily FROMLONLAT 15 37 BYBOX 400 400 km ASC WITHCOORD WITHDIST count 1 any
1) 1) "Palermo"
   2) "190.4424"
   3) 1) "13.36138933897018433"
      2) "38.11555639549629859"
```

look at the difference, count 1 will return the closest one
count 1 any will just return the first one that was found

Reply:
But the are always in the search criteria, that's the point of the command :)

Reply:
Oh now I see. Very nice example, thank you!
Can you include this into tests if it is not here yet?